systems/notification/api-gateway/pkg/rest/router.go

```diff
@@ -118,20 +118,26 @@ func formatDoc(summary string, description string) []fizz.OperationOption {
 
 func (r *Router) sendEmailHandler(c *gin.Context, req *SendEmailReq) (message emailPkg.SendEmailResponse, err error) {
 	payload := emailPkg.SendEmailRequest{
-		To:      req.To,
-		Subject: req.Subject,
-		Body:    req.Body,
-		Values:  req.Values,
+		To:           req.To,
+		TemplateName: req.TemplateName,
+		Values:       make(map[string]string),
 	}
+
+	// Convert map[string]interface{} to map[string]string
+	for key, value := range req.Values {
+		if strValue, ok := value.(string); ok {
+			payload.Values[key] = strValue
+		}
+	}
+
 	res, err := r.clients.m.SendEmail(&payload)
 	if err != nil {
 		return emailPkg.SendEmailResponse{}, err
 	}
 
-
 	return emailPkg.SendEmailResponse{
 		Message: res.Message,
-		MailId: res.MailId,
+		MailId:  res.MailId,
 	}, nil
 }
 
```

Comment:
Same goes for: 
```go
func (r *Router) getEmailByIdHandler(c *gin.Context, req *GetEmailByIdReq) (message emailPkg.GetEmailByIdResponse, err error) {
	payload := emailPkg.GetEmailByIdRequest{
		MailId: req.MailerId,
	}
	res, err := r.clients.m.GetEmailById(&payload)
	if err != nil {
		return emailPkg.GetEmailByIdResponse{}, err
	}

	return emailPkg.GetEmailByIdResponse{
		MailId: res.MailId,
		To:      res.To,
		Subject: res.Subject,
		Body:    res.Body,		
	}, nil

}
```

That part : 
```go
return emailPkg.GetEmailByIdResponse{
		MailId: res.MailId,
		To:      res.To,
		Subject: res.Subject,
		Body:    res.Body,		
	}, nil

}
```
is not needed.  `res` is already a `pb.GetEmailByIdResponse`.